src/webgpu/api/operation/compute/overrides.spec.ts

```diff
@@ -0,0 +1,409 @@
+      
+      @compute @workgroup_size(1) fn main() {
+          buf.data[0] = u32(c0);
+          buf.data[1] = u32(c1);
+          buf.data[2] = u32(c2);
+          buf.data[3] = u32(c3);
+          buf.data[4] = u32(c4);
+          buf.data[5] = u32(c5);
+          buf.data[6] = u32(c6);
+          buf.data[7] = u32(c7);
+          buf.data[8] = u32(c8);
+          buf.data[9] = u32(c9);
+          buf.data[10] = u32(c10);
+      }
+    `
+    );
+  });
+
+g.test('numeric_id')
+  .desc(`Test that correct values are used for constants with numeric id.`)
+  .fn(async t => {
+    t.ExpectShaderOutputWithConstants(
+      new Uint32Array([1, 2, 3]),
+      {
+        1001: 1,
+        1: 2,
+        // 1003 is using default value
+      },
+      `
+        @id(1001) override c1: u32;            // some big numeric id
+        @id(1) override c2: u32 = 0u;          // id == 1 might collide with some generated constant id
+        @id(1003) override c3: u32 = 3u;       // default
+        
+        struct Buf {
+            data : array<u32, 3>
+        }
+        
+        @group(0) @binding(0) var<storage, read_write> buf : Buf;
+        
+        @compute @workgroup_size(1) fn main() {
+            buf.data[0] = c1;
+            buf.data[1] = c2;
+            buf.data[2] = c3;
+        }
+      `
+    );
+  });
+
+g.test('precision')
+  .desc(`Test that float number precision is preserved for constants.`)
+  .fn(async t => {
+    const c1 = 3.14159;
+    const c2 = 3.141592653589793238;
+    t.ExpectShaderOutputWithConstants(
+      new Float32Array([c1, c2]),
+      {
+        c1,
+        c2,
+      },
+      `
+        override c1: f32;
+        override c2: f32;
+        
+        struct Buf {
+            data : array<f32, 2>
+        }
+        
+        @group(0) @binding(0) var<storage, read_write> buf : Buf;
+        
+        @compute @workgroup_size(1) fn main() {
+            buf.data[0] = c1;
+            buf.data[1] = c2;
+        }
+      `
+    );
+  });
+
+g.test('workgroup_size')
+  .desc(`Test that constants can be used as workgroup size correctly.`)
+  .params(u =>
+    u //
+      .combine('x', [3, 16, 64])
+  )
+  .fn(async t => {
+    const { x } = t.params;
+    t.ExpectShaderOutputWithConstants(
+      new Uint32Array([x]),
+      {
+        x,
+      },
+      `
+        override x: u32;
+
+        struct Buf {
+            data : array<u32, 1>
+        }
+        
+        @group(0) @binding(0) var<storage, read_write> buf : Buf;
+        
+        @compute @workgroup_size(x) fn main(
+            @builtin(local_invocation_id) local_invocation_id : vec3<u32>
+        ) {
+            if (local_invocation_id.x >= x - 1) {
+                buf.data[0] = local_invocation_id.x + 1;
+            }
+        }
+      `
+    );
+  });
+
+g.test('shared_shader_module')
+  .desc(
+    `Test that when the same shader module is shared by different pipelines, the correct constant values are used.`
```

Comment:
side note: when we do operation tests for render pipelines, would be good to have an override that's used by both the vertex and fragment entry points and override it to a different value in each stage. Should put this in a TODO somewhere or in a github issue to track it.

Reply:
Add a `api/operation/render_pipeline/overrides.spec.ts` placeholder.